docs/features/logforwarding/outputs/lokistack-forwarding.adoc

```diff
@@ -0,0 +1,450 @@
+----
+<1> `serviceAccount.name` must have permissions to both *collect* AND *write* app and infra logs
+<2> `lokiStack.target` name and namespace must match your loki instance name
+<3> TLS configuration `key` and `configMapName` uses the existing openshift service config map
+
+
+== Alternatively using custom outputs and pipelines
+
+=== ClusterLogForwarder
+* Create a ClusterLogForwarder CR to forward logs to individual tenant endpoints in LokiStack
++
+.ClusterLogForwarder with custom pipelines, filtered and based on log_type
+[source,yaml]
+----
+    apiVersion: observability.openshift.io/v1
+    kind: ClusterLogForwarder
+    metadata:
+      name: my-forwarder
+      namespace: openshift-logging
+    spec:
+      serviceAccount:
+        name: logging-admin
+      outputs:
+      - name: loki-app
+        type: loki
+        loki:
+          url: https://logging-loki-gateway-http.openshift-logging.svc:8080/api/logs/v1/application
+          authentication:
+            token:
+              from: serviceAccount
+        tls:
+          ca:
+            key: service-ca.crt
+            configMapName: openshift-service-ca.crt
+      - name: loki-infra
+        type: loki
+        loki:
+          url: https://logging-loki-gateway-http.openshift-logging.svc:8080/api/logs/v1/infrastructure
+          authentication:
+            token:
+              from: serviceAccount
+        tls:
+          ca:
+            key: service-ca.crt
+            configMapName: openshift-service-ca.crt
+      - name: loki-audit
+        type: loki
+        loki:
+          url: https://logging-loki-gateway-http.openshift-logging.svc:8080/api/logs/v1/audit
+          authentication:
+            token:
+              from: serviceAccount
+        tls:
+          ca:
+            key: service-ca.crt
+            configMapName: openshift-service-ca.crt
+      filters:
+      - name: my-multi
+        type: detectMultilineException
+      - name: my-parse
+        type: parse
+      - name: my-labels
+        type: openshiftLabels
+        openshiftLabels:
+          foo: bar
+      pipelines:
+      - name: send-app-logs
+        inputRefs:
+        - application
+        outputRefs:
+        - loki-app
+        filterRefs:
+        - my-multi
+        - my-parse
+        - my-labels
+      - name: send-infra-logs
+        inputRefs:
+        - infrastructure
+        outputRefs:
+        - loki-infra
+        filterRefs:
+        - my-multi
+        - my-parse
+        - my-labels
+      - name: send-audit-logs
+        inputRefs:
+        - audit
+        outputRefs:
+        - loki-audit
+        filterRefs:
+        - my-multi
+        - my-parse
+        - my-labels
+----
+For the internal loki gateway service, we use the url format *https://<service-name>.<namespace>.svc:8080/api/logs/v1/<log_type>*
+
+== Alternative Authentication
+
+=== Token Secret
+* Create a secret containing the service account bearer token
++
+.Secret from a service account token
```

Comment:
So this says, give populate my secret with a token for a certain SA.  I thought there was a flip of this where you annotate an SA and tell it to create and populate a secret.  This seems like a more concise workflow

Reply:
I've removed this section, as this type of authentication is only valid for HCP as far as I know.